core/tests/protractor_utils/TopicEditorPage.js

```diff
@@ -258,30 +260,93 @@ var TopicEditorPage = function() {
+    await waitFor.elementToBeClickable(reassignSkillButton,
+      'Reassign skill button taking too long to be clickable');
+    await reassignSkillButton.click();
+  };
+
+  this.expectSubtopicWithIndexToHaveSkills = async function(
+      subtopicIndex, skillNames) {
+    const assignedSkillDescriptions = (
+      subtopicColumns.get(subtopicIndex).all(
+        by.css('.protractor-test-subtopic-skill-description')));
+    const assignedSkillsLength = await assignedSkillDescriptions.count();
+
+    expect(skillNames.length).toEqual(assignedSkillsLength);
+
+    for (let i = 0;i < assignedSkillsLength;i++) {
+      const skillDescription = await assignedSkillDescriptions.get(i).getText();
+      expect(skillDescription).toEqual(skillNames[i]);
+    }
+  };
+
+  this.dragSkillFromSubtopicToSubtopic = async function(
+      fromSubtopicIndex, toSubtopicIndex, skillDescription) {
+    const fromSubtopicColumn = subtopicColumns.get(fromSubtopicIndex);
+    const assignedSkills = fromSubtopicColumn.all(
+      by.css('.protractor-test-subtopic-skill-description'));
+    const assignedSkillsLength = await assignedSkills.count();
+    var toMoveSkillIndex = -1;
+    for (var i = 0;i < assignedSkillsLength;i++) {
+      if (skillDescription === await assignedSkills.get(i).getText()) {
+        toMoveSkillIndex = i;
+        break;
+      }
+    }
+    expect(toMoveSkillIndex).not.toEqual(-1);
+    const assignedSkillToMove = assignedSkills.get(toMoveSkillIndex);
+    const toSubtopicColumn = subtopicColumns.get(toSubtopicIndex);
+    await dragAndDrop(assignedSkillToMove, toSubtopicColumn);
+  };
+
+  this.expectUncategorizedSkillsToBe = async function(skillDescriptions) {
+    var uncategorizedSkills = (
+      element.all(by.css('.protractor-test-uncategorized-skill-card')));
+
+    await waitFor.visibilityOf(await uncategorizedSkills.first(),
+      'Uncategorized skills taking too long to appear.');
+
+    for (var i = 0;i < await uncategorizedSkills.count();i++) {
+      expect(skillDescriptions[i]).toEqual(
+        await uncategorizedSkills.get(i).getText());
+    }
   };
 
   this.dragSkillFromSubtopicToUncategorized = async function(
-      skillCardIndex) {
-    var skillNamesElems = element.all(
-      by.css('.protractor-test-assigned-skill-card-text'));
-    var toMove = await skillNamesElems.get(skillCardIndex);
-    await dragAndDrop(toMove, uncategorizedSkillsContainer);
+      subtopicIndex, skillDescription) {
+    const fromSubtopicColumn = subtopicColumns.get(subtopicIndex);
+    const assignedSkills = fromSubtopicColumn.all(
+      by.css('.protractor-test-subtopic-skill-description'));
+    const assignedSkillsLength = await assignedSkills.count();
+    var toMoveSkillIndex = -1;
+    for (var i = 0;i < assignedSkillsLength;i++) {
+      if (skillDescription === await assignedSkills.get(i).getText()) {
+        toMoveSkillIndex = i;
+        break;
+      }
+    }
+    expect(toMoveSkillIndex).not.toEqual(-1);
```

Comment:
This code gets duplicated a lot. Can you pull it into a function?

Reply:
Done.